shared/localization/format.js

```diff
@@ -354,6 +359,10 @@ function hasLocale(requestedLocale) {
   return hasIntlSupport && hasMessages;
 }
 
+function getCanonicalLocales() {
+  return DEFAULT_LOCALES;
```

Comment:
cc @brendankenny the swap locale ui shouldn't show the "duplicate" locales

Reply:
> cc @brendankenny the swap locale ui shouldn't show the "duplicate" locales

hmm, it seems like there's a bug somewhere in #13146 then? `getAvailableLocales` should be returning the union of `DEFAULT_LOCALES` and `Object.keys(LOCALE_MESSAGES)`. If `locales.js` is shimmed to `export default {}`, shouldn't `const LOCALE_MESSAGES = require('./locales.js');` be just `{}`, so the union of `DEFAULT_LOCALES` and `Object.keys(LOCALE_MESSAGES)` would be just `DEFAULT_LOCALES`?

Reply:
Ah, so it was fine in the viewer src, but the issue was that the jest test evaluates this in node as-is, and was including tons of extra locales failing the assertion.

Reply:
if it's in the pptr test it should be ok, then? Otherwise there's not much point to most of `getAvailableLocales` and it should just be updated to return `Object.keys(LOCALE_MESSAGES)`.

Reply:
It still serves a purpose here, no? https://github.com/GoogleChrome/lighthouse/blob/1fe1ce866456c38a680db6d46ca5682d3084f602/lighthouse-core/lib/i18n/i18n.js#L137

> if it's in the pptr test it should be ok, then? 

`expect(resultBeforeSwap.options).toEqual(getCanonicalLocales());` would fail if it instead called `getAvailableLocales` (even if swap-locale-feature used it too), because only the viewer build causes `getAvailableLocales` to return just the "canonical" locales

This is just an issue in the test... as long as this module shim for `locales.js` exists, could remove the second "get locales" function if the test just asserts `length > 10` or something.

But I think the split between "canonical" (is there a better word) and "all" locales makes it more apparent what's going on.

Reply:
> It still serves a purpose here, no?
> 
> https://github.com/GoogleChrome/lighthouse/blob/1fe1ce866456c38a680db6d46ca5682d3084f602/lighthouse-core/lib/i18n/i18n.js#L137

that's what I meant. If `getAvailableLocales` returns `Object.keys(LOCALE_MESSAGES)` that line should be fine.

> `expect(resultBeforeSwap.options).toEqual(getCanonicalLocales());` would fail if it instead called `getAvailableLocales` (even if swap-locale-feature used it too), because only the viewer build causes `getAvailableLocales` to return just the "canonical" locales

but `viewer-test-pptr.js` is running against the viewer build, isn't it?

> But I think the split between "canonical" (is there a better word) and "all" locales makes it more apparent what's going on.

yeah, I'm coming around to this view as well, just need to document the methods to make sure it's clear.

Reply:
Gave updating these methods a shot, what do you think?

> but viewer-test-pptr.js is running against the viewer build, isn't it?

Sorry, I've lost track of the confusion here :)